warehouse/cli/tuf.py

```diff
@@ -0,0 +1,171 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+
+from securesystemslib.exceptions import StorageError
+from securesystemslib.interface import generate_and_write_ed25519_keypair
+
+from warehouse.cli import warehouse
+from warehouse.packaging.utils import render_simple_detail
+from warehouse.tuf.tasks import (
+    add_hashed_targets as _add_hashed_targets,
+    bump_bin_n_roles as _bump_bin_n_roles,
+    bump_snapshot as _bump_snapshot,
+    init_repository as _init_repository,
+    init_targets_delegation as _init_targets_delegation,
+)
+
+
+@warehouse.group()  # pragma: no-branch
+def tuf():
+    """
+    Manage Warehouse's TUF state.
+    """
+
+
+@tuf.group()
+def dev():
+    """
+    TUF Development purposes commands
+    """
+
+
+@dev.command()
+@click.pass_obj
+@click.option("--name", "name_", help="The name of the TUF role for this keypair")
+@click.option("--path", "path_", help="The basename of the Ed25519 keypair to generate")
+def keypair(config, name_, path_):
+    """
+    Generate a new TUF keypair for top-level roles.
```

Comment:
Any keypair really, also for non-top-level roles, bins and bin-n, right?

I can quickly polish these docstrings, if you want.

Reply:
Yes. It can generate any key keypair